library/std/src/panicking.rs

```diff
@@ -261,6 +263,14 @@ pub mod panic_count {
         })
     }
 
+    pub fn set_always_abort() {
+        GLOBAL_PANIC_COUNT.fetch_add(1, Ordering::Relaxed); // it will always be nonzero, now
+        LOCAL_PANIC_COUNT.with(|c| {
```

Comment:
Why only set this on the current thread instead of for the entire process?

Reply:
> Why only set this on the current thread instead of for the entire process?

This is a thinko on my part.  I was thinking about the post-`fork` case where of course there is only one thread.  But `panic::always_panic` may be used in other situations and should be entirely global.  That means that the magic bit needs to be in GLOBAL_PANIC_COUNT, not in LOCAL_PANIC_COUNT.  I will rework this.

And also add a test :-).

Reply:
> Why only set this on the current thread instead of for the entire process?

Fixing this involved strengthening the `Ordering` of the reads of the global panic count, to `Acquire`.  The writes are still `Relaxed`.  I hope this is OK.  I think this is a necessary consequence of having `panic::always_abort` be global.  Having it be local would be an alternative possibility (though I would have to check what happens if a thread with a local panic count ends...)